### PR TITLE
fix folder creation errors on Windows; disallow empty input in new folder modals

### PR DESCRIPTION
**Pull request by sharon-wang**
- addresses: #5195 
- since I was in the area, I added a fix for https://github.com/posit-dev/positron/issues/2777 to disallow the modal accept if no folder name or git repo was entered

### QA Notes

I've tested on Windows 11 Desktop, Mac Desktop, Mac Server Web and Linux REH-Web.

Each of the following modals should now work for creating a new folder/project on Windows.
- New Folder
- New Folder from Git
- New Project

Clicking <kbd>OK</kbd> in New Folder and New Folder from Git modals without entering a folder name or git repo now shows and error and doesn't proceed unless the user tries again with the required input.

### CODE REVIEW

**Comment by sharon-wang** [comment]
Kicked off a Windows Test Suite run: https://github.com/posit-dev/positron/actions/runs/11581469429

File: src/vs/workbench/browser/positronComponents/positronModalDialog/components/fileInputValidators.ts [hunk 1]
```diff
@@ -113,6 +113,16 @@ export async function checkIfURIExists(path: URI, fileService: IFileService): Pr
...
 
+/**
+ * Check if the input is empty.
+ * @param input The input to check if it is empty.
+ * @returns Whether the input is empty.
+ */
+export function isInputEmpty(input: string | number): boolean {
+	const inputString = typeof input === 'number' ? input.toString() : input;
+	return inputString.trim() === '';
+}
```
Comment (softwarenerd) [inline src/vs/workbench/browser/positronComponents/positronModalDialog/components/fileInputValidators.ts:124]: Just a thought:

```
export function isInputEmpty(input: string | number): boolean {
	if (typeof input === 'number') {
		return false;
	} else {
		return input.trim() === '';
	}
}
```

**Review by softwarenerd — DISMISSED** [review]
LG!

File: src/vs/workbench/browser/positronComponents/positronModalDialog/components/fileInputValidators.ts (hunk 1 above)
Comment (sharon-wang) [reply src/vs/workbench/browser/positronComponents/positronModalDialog/components/fileInputValidators.ts:123]: Thank you! I went with the second one :)

**Pull request merged**